File: apps/api/src/lib/cf.ts
const CF_RE = /^[A-Z0-9]{16}$/i;

export function normalizeFiscalCode(cf: string): string {
	return cf.trim().toUpperCase().replace(/\s/g, '');
}

export function isValidFiscalCodeFormat(cf: string): boolean {
	const n = normalizeFiscalCode(cf);
	return CF_RE.test(n);
}

File: apps/python-api/src/yourmedportal_api/lib/cf.py
import re

CF_RE = re.compile(r"^[A-Z0-9]{16}$", re.IGNORECASE)


def normalize_fiscal_code(cf: str) -> str:
    return cf.strip().upper().replace(" ", "")


def is_valid_fiscal_code_format(cf: str) -> bool:
    n = normalize_fiscal_code(cf)
    return bool(CF_RE.match(n))
